src/expression.d

```diff
@@ -466,22 +500,21 @@ extern (C++) Expression resolvePropertiesX(Scope* sc, Expression e1, Expression
         else if (e1.op == TOKdotvar)
         {
             // Check for reading overlapped pointer field in @safe code.
-            VarDeclaration v = (cast(DotVarExp)e1).var.isVarDeclaration();
-            if (v && v.overlapped && sc.func && !sc.intypeof)
-            {
-                AggregateDeclaration ad = v.toParent2().isAggregateDeclaration();
-                if (ad && e1.type.hasPointers() && sc.func.setUnsafe())
-                {
-                    e1.error("field %s.%s cannot be accessed in @safe code because it overlaps with a pointer", ad.toChars(), v.toChars());
-                    return new ErrorExp();
-                }
-            }
+            if (checkOverlappedPointer(sc, e1, "field %s.%s cannot be accessed in @safe code because it overlaps with a pointer"))
```

Comment:
Can you add the name of the other field that is considered a pointer to the message? E.g.
`"...because it overlaps with U.sysDg which is a pointer. Reinterpreting pointers in not allowed in @safe code."` for the test case below.